src/protocol/data/redis/compose.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <buffer/cc_dbuf.h>
+#include <cc_define.h>
+#include <cc_metric.h>
+
+#include <stdint.h>
+
+/*          name                    Type            description */
+#define COMPOSE_REQ_METRIC(ACTION)                                          \
+    ACTION( request_compose,        METRIC_COUNTER, "# requests composed"  )\
+    ACTION( request_compose_ex,     METRIC_COUNTER, "# composing error"    )
+
+/*          name                    Type            description */
+#define COMPOSE_RSP_METRIC(ACTION)                                          \
+    ACTION( response_compose,       METRIC_COUNTER, "# responses composed" )\
+    ACTION( response_compose_ex,    METRIC_COUNTER, "# rsp composing error")
+
+typedef struct {
+    COMPOSE_REQ_METRIC(METRIC_DECLARE)
+} compose_req_metrics_st;
+
+typedef struct {
+    COMPOSE_RSP_METRIC(METRIC_DECLARE)
+} compose_rsp_metrics_st;
+
+typedef enum compose_rstatus {
+    COMPOSE_OK          = 0,
+    COMPOSE_EUNFIN      = -1,
+    COMPOSE_ENOMEM      = -2,
+    COMPOSE_EINVALID    = -3,
+    COMPOSE_EOTHER      = -4,
+} compose_rstatus_t;
```

Comment:
Yea you are right, and I noticed that too; let me create an issue to change this for all protocols.